.circleci/config.yml

```diff
@@ -418,14 +418,14 @@ commands:
     steps:
       - restore_cache:
           keys:
-            - rust-target-v2-<< parameters.platform >>-{{ checksum "Cargo.lock" }}
+            - &target-cache-key rust-target-v3-<< parameters.platform >>-{{ .Branch }}-{{ checksum "Cargo.lock" }}
 
       - run:
           command: cargo xtask << parameters.command >> << parameters.options >>
           working_directory: << parameters.working_directory >>
 
       - save_cache:
```

Comment:
could we just wrap this in condition to save cache from `main` branch?

```yaml
     - when:
          condition:
            matches: // or equal
              pattern: "/main/"
              value: << get target branch >> 
          steps:
            - save_cache:
                key: *target-cache-key
                paths:
                  - target/
```

Reply:
*not a circle ci expert so unsure if it is even possible but I think we might be able to get some condition to match the main branch OR something like tag value

Reply:
Yeah, I'm no expert either, so I decided to keep it simple and disable caching instead of having to wait for the next release to find out if we did it correctly.